### ISSUE TITLE
Pen registration based on measured pen diameter

### ISSUE THREAD

**Issue by Shrinks99** [evil-mad/axidraw#176]
One of the more annoying parts of plotting with multiple brands of pens is ensuring they are aligned.  Most folks seem to get around this by plotting registration patterns and shifting the paper around until it lines up with the pen center.  This process is error prone and inexact, but also seems quite solvable in software!

### Proposed solution

Currently I don't have much insight into where exactly the plotter considers the exact zero to be.  That said, assuming we know the diameter of the pen (and at least have some sort of starting value to measure against) it should be fairly easy to calculate the radius and offset the difference?  Anyone with a pair of calipers should be able to get these measurements and create more accurate plots!

**Comment by oskay** [comment]
> Most folks seem to get around this by plotting registration patterns and shifting the paper around until it lines up with the pen center. 

This isn't a method that is very easy to use; the "walk" offsets in the AxiDraw software can be used to offset the starting point.  If half of the pen diameter difference is sufficiently accurate for your needs, then that should be straightforward.

**Comment by Shrinks99** [comment]
Yeah!  Aware of the walk parameter, also perhaps shouldn't have asserted that's what most folks are doing... It's what a few articles online mentioned doing.

In leu of a more streamlined solution I'll definitely be applying the offsets in software, either with the walk offset or in Inkscape.

**Comment by oskay** [comment]
There is, unfortunately, often much more variation in effective pen tip position than from just the barrel diameter, so we have been hesitant to make accessories or software that implies that merely correcting for barrel diameter is sufficient to get perfect precision.

**Comment by Shrinks99** [comment]
Hmm... I'm intrigued!  Admitidly I'm not the most experienced with plotters, I'm sure you'd have better insights than I would as to what those other dimensions of difficulty are?  If you know of any further reading on the subject I'd love to know about it!

**Comment by Shrinks99** [comment]
Well, after breaking out the callipers and trying the above today it seems you're right and it is indeed not that simple! The barrel dimeter of all the pens I tested was non-uniform, meaning alignment a bit more nuanced than I initially expected.

Happy to close this as unplanned if you'd like to keep open issues free of clutter, but at least there is one now to show it's been thought of!
